Pull access denied for microsoft/aspnetcore-build while following the exact steps in tutorial #82135

_Originally posted by @Minetorpia in https://github.com/MicrosoftDocs/azure-docs/issues/82135_

I followed every step in the tutorial, but sadly building the image fails. I tried Googling, but didn't find anything on how to fix this.

```
Step 1/20 : FROM microsoft/aspnetcore-build AS build-env
pull access denied for microsoft/aspnetcore-build, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
##[error]pull access denied for microsoft/aspnetcore-build, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
##[error]The process '/usr/bin/docker' failed with exit code 1
Finishing: Build and push an image to container registry
```


---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4cfc7aa5-d9ff-ad64-d26e-d2bd10470025
* Version Independent ID: 9b4f02ea-dab9-f613-5345-496435ab2408
* Content: [Deploy containerized applications to Web App for Containers - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/apps/cd/deploy-docker-webapp?view=azure-devops&tabs=dotnet-core)
* Content Source: [docs/pipelines/apps/cd/deploy-docker-webapp.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/master/docs/pipelines/apps/cd/deploy-docker-webapp.md)
* Product: **devops**
* Technology: **devops-cicd-apps**
* GitHub Login: @RoopeshNair
* Microsoft Alias: **ronai**

Today I tried to follow steps from here: 
https://learn.microsoft.com/en-us/azure/devops/pipelines/apps/cd/deploy-docker-webapp?view=azure-devops&tabs=dotnet-core%2Cclassic
but ends up getting the same error as follows:
2023-04-12T08:27:27.4323436Z Step 1/20 : FROM microsoft/aspnetcore-build AS build-env
2023-04-12T08:27:28.3991274Z pull access denied for microsoft/aspnetcore-build, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
2023-04-12T08:27:28.4101428Z ##[error]pull access denied for microsoft/aspnetcore-build, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
2023-04-12T08:27:28.4141590Z ##[error]The process '/usr/bin/docker' failed with exit code 1
2023-04-12T08:27:28.4218355Z ##[section]Finishing: Build and push an image to container registry

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,5 +1,5 @@
 # First stage of multi-stage build
-FROM microsoft/aspnetcore-build AS build-env
+FROM mcr.microsoft.com/dotnet/framework/sdk AS build-env
 WORKDIR /app
 
 # copy the contents of agent working directory on host to workdir in container
@@ -11,7 +11,7 @@ RUN dotnet build -c Release
 RUN dotnet publish -c Release -o out
 
 # Second stage - Build runtime image
-FROM microsoft/aspnetcore
+FROM mcr.microsoft.com/dotnet/aspnet
 WORKDIR /app
 COPY --from=build-env /app/out .
 ENTRYPOINT ["dotnet", "pipelines-dotnet-core-docker.dll"]

